crates/sp-domains/src/fraud_proof.rs

```diff
@@ -16,7 +16,10 @@ pub enum ExecutionPhase {
     /// Executes the `initialize_block` hook.
     InitializeBlock { domain_parent_hash: H256 },
     /// Executes some extrinsic.
-    ApplyExtrinsic(u32),
+    ApplyExtrinsic {
+        extrinsic_index: u32,
+        primary_hash: H256,
```

Comment:
i'm not sure about this one TBH. The verifier should have this info already. If not they should be able to get this from the receipts on the state through `primary_hash` no?

Reply:
This reminds me that we may need to verify the integrity of `primary_parent_hash` and `primary_hash` to ensure `primary_parent_hash` is indeed `primary_hash`'s parent hash.

Otherwise, since `primary_parent_hash` is used to fetch the runtime code, the attacker can set it to an old value and construct a state transition proof that is valid for the old runtime but invalid for the runtime that the receipt targetted and still pass the verification.

Reply:
> There are potentially multiple receipts at a certain height,

True but there will only one canonical primary block and this can be retrieved from the embed primary client or through `PrimaryBlockHash` on the pallet-receipts since we already know the block number.

Reply:
> This reminds me that we may need to verify the integrity of primary_parent_hash and primary_hash to ensure primary_parent_hash is indeed primary_hash's parent hash.

yes or we can just extract this info from the client and verification is done implicitly.

Reply:
It's now retrieved from `PrimaryBlockHash` in the lastest push, but `pre_state_root` remains unchanged as some non-trivial test fixes are needed to refactor it, a TODO is added which could be probably resolved when the fraud proof tests are improved.